### PR TITLE
[src] Fixing the scaling bug of the general-dropout component

### PR DESCRIPTION
**Pull request by GaofengCheng**
fix the scale factor bug for sequence-shared dropout

related to #2814

### CODE REVIEW

**Comment by GaofengCheng** [comment]
@danpovey 
Hi Dan, you may be interested in this PR. This may solve a bug in GeneralDropoutComponnet, which will lead the breakdown of training (when we set a small dropout value like 0.1). This does not influence the continuous dropout.

**Comment by danpovey** [comment]
Hm, interesting.
I suspect in setups with batch-norm (which is most setups),this would not have made much of a difference in practice.  But it might depend where exactly the batchnorm is.
I'll merge it.
